cabal-install-solver/src/Distribution/Solver/Types/PkgConfigDb.hs

```diff
@@ -65,9 +67,14 @@ readPkgConfigDb verbosity progdb = handle ioErrorHandler $ do
         -- The output of @pkg-config --list-all@ also includes a description
         -- for each package, which we do not need.
         let pkgNames = map (takeWhile (not . isSpace)) pkgList
-        pkgVersions <- lines <$> getProgramOutput verbosity pkgConfig
-                                   ("--modversion" : pkgNames)
-        (return . pkgConfigDbFromList . zip pkgNames) pkgVersions
+        (pkgVersions, _errs, exitCode) <-
+                     getProgramInvocationOutputAndErrors verbosity
+                       (programInvocation pkgConfig ("--modversion" : pkgNames))
+        print (pkgVersions, exitCode)
+        case exitCode of
+          ExitSuccess -> (return . pkgConfigDbFromList . zip pkgNames) (lines pkgVersions)
+          -- if there's a single broken pc file the above fails, so we fall back into calling it individually
+          _ -> pkgConfigDbFromList . catMaybes <$> mapM (getIndividualVersion pkgConfig) pkgNames
```

Comment:
Why not just always use this code path? Is it that much slower than getting all packages in one call?

Reply:
Depends on the size of the database :-)